cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -137,28 +130,31 @@ public void destroy() {
   }
 
   private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
+                                         CConfiguration cConf) {
     Preconditions.checkState(port > 0, "Port number is invalid.");
+    int numPartitions = cConf.getInt(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG);
     Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
 
     Properties prop = new Properties();
+
     prop.setProperty("broker.id", Integer.toString(brokerId));
     if (hostname != null) {
       prop.setProperty("host.name", hostname);
     }
     prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("socket.send.buffer.bytes", "1048576");
-    prop.setProperty("socket.receive.buffer.bytes", "1048576");
-    prop.setProperty("socket.request.max.bytes", "104857600");
-    prop.setProperty("log.dir", logDir);
-    prop.setProperty("num.partitions", Integer.toString(numPartitions));
-    prop.setProperty("log.retention.hours", "24");
-    prop.setProperty("log.flush.interval.messages", "10000");
-    prop.setProperty("log.flush.interval.ms", "1000");
-    prop.setProperty("log.segment.bytes", "536870912");
     prop.setProperty("zookeeper.connect", zkConnectStr);
-    prop.setProperty("zookeeper.connection.timeout.ms", "1000000");
-    prop.setProperty("default.replication.factor", Integer.toString(replicationFactor));
+
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka.*)$");
+    for (Map.Entry<String, String> pair : propConfigs.entrySet()) {
```

Comment:
See if it makes sense to use:
```java
ConfigurationUtil#getNamedConfigurations(cConf, "kafka.");
```

It will abstract away the getValByRegex as well as having to do key.substring(6).